### ISSUE TITLE
Some examples are not tested properly

### ISSUE THREAD

**Issue by Sliman4** [near/near-sdk-rs#1190]
Just cloned the repo, tried to follow `CONTRIBUTING.md`, here's what I noticed:
- When you run `./run-tests.sh`, it doesn't work because examples .wasm files are not built yet, and there's no notice that you should run `examples/build_all.sh` beforehand. Not sure what's the best way to fix it, since there is a `build_all_docker.sh` that may or may not be used in some cases.
- Some examples are not tested by CI, for example, `examples/non-fungible-token` fails to build tests because it doesn't use `unit-testing` feature of SDK, and `examples/versioned` has the same problem + uses deprecated `UnorderedMap`.

**Comment by frol** [comment]
@Sliman4 Would you be open to addressing these? I am happy to review and merge PRs

**Comment by Sliman4** [comment]
Some examples seem to be intentionally removed from gh actions (not sure why, but I guess there was a good reason), and I believe `ci-test.sh` is not referenced from anywhere, unless I don't understand something. I think it would be better if it was done by someone with more knowledge about CI, and specifically workflows of this repo, as I don't feel confident that I understand how and what is supposed to be tested, and don't have time to go through relevant commits, issues, and PRs

**Comment by frol** [comment]
It would be great to make sure we test all examples as part of CI jobs. I marked this issue as "good first issue" as I believe someone with a bit of GitHub Actions and bash experience should be successful at fixing it.

**Comment by shivam6862** [comment]
I am applying to this issue via [OnlyDust-platform](https://app.onlydust.com/p/near-sdk-rs).

Hi, I can work on this. I am a developer with three years of experience. I will complete it in 3 working days

### Plan
- Ensure .wasm files are built before running tests: Modify run-tests.sh to include a step that builds the examples before running the tests.
- Update CI in test_examples: Modify the CI configuration to include all examples, including those not currently tested.
- Fix examples that fail to build: Address the issues in examples/non-fungible-token and examples/versioned.

**Comment by renzobanegass** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/near-sdk-rs).

### My background and how it can be leveraged
Hello! I'm a Software Engineer with over 5 years of experience and also part of the Dojo Coding community, I had worked with CI/CD in the past and have experience developing and fixing pipelines, I'm new to Web3 and this would be my first contribution!

### How I plan on tackling this issue
How would you approach the problem?
Review the Existing Setup:

Clone the repository and go through the CONTRIBUTING.md, existing CI workflow files, and the example projects to understand the current setup and the issues mentioned.
Update Documentation:

Modify the CONTRIBUTING.md to clearly instruct contributors to run examples/build_all.sh before ./run-tests.sh. This will prevent confusion and ensure that the examples are built before testing.
Enhance CI Workflow:

Edit the .github/workflows/ci.yml (or the relevant workflow file) to include steps for building and testing all examples. This ensures that all examples are automatically tested in the CI pipeline.

Add Missing Tests:

Ensure examples/non-fungible-token and examples/versioned have proper test coverage. This may involve creating or updating test files in these directories.
Verify that these examples are included in the CI workflow and pass the tests.
Refactor Deprecated Features:

Update examples to replace deprecated features like UnorderedMap with supported data structures

Testing and Validation:

Run all tests locally to ensure they pass after the changes.
Push the changes to a forked repository and create a pull request.
Ensure the CI pipeline runs successfully on the pull request and all tests pass.
Create a Pull Request:

Provide a detailed description of the changes, including links to the modified files, the rationale behind the changes, and how they address the issue.
Be available to address any feedback or questions from the maintainers and contributors.

**Comment by Gift-Naomi** [comment]
@frol can I please work on this?

**Comment by frol** [comment]
@renzobanegass Please, go ahead!

**Comment by renzobanegass** [comment]
Hey @frol I made the changes and all is working properly but I had an issue with the factory-contract tests, before making any changes I ran the tests and it failed. I had to ignore it to continue developing but now that I finished it is still failing. 

The res.is_success() assertion is failing in 
test_deploy_status_message::_factory_contract_high_level_expects
test_deploy_status_message::_factory_contract_low_level_expects

Any idea on how to handle this? 
The rest is ready for a PR, it seems the examples were taken out off of the CI for a reason so I want to run the pipeline to check what happened.
